# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.h
# Commit: 77eda17d0795
# Full Hash: 77eda17d0795e32bd6bf224832b922d8e57aedca
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-15 21:21:06
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r 358f0cf9a8d8 -r 77eda17d0795 dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.h	Tue Apr 15 09:47:38 2025 +0000
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_RemoteWorkerDebuggerManagerParent_h
+#define mozilla_dom_RemoteWorkerDebuggerManagerParent_h
+
+#include "mozilla/dom/PRemoteWorkerDebuggerManagerParent.h"
+#include "mozilla/dom/PRemoteWorkerDebuggerParent.h"
+#include "mozilla/ipc/Endpoint.h"
+
+namespace mozilla::dom {
+
+class RemoteWorkerDebuggerManagerParent final
+    : public PRemoteWorkerDebuggerManagerParent {
+  friend class PRemoteWorkerDebuggerManagerParent;
+
+ public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(RemoteWorkerDebuggerManagerParent,
+                                        final)
+
+  RemoteWorkerDebuggerManagerParent();
+
+  mozilla::ipc::IPCResult RecvRegister(
+      const RemoteWorkerDebuggerInfo& aDebuggerInfo,
+      mozilla::ipc::Endpoint<PRemoteWorkerDebuggerParent>&& aParentEp);
+
+ private:
+  ~RemoteWorkerDebuggerManagerParent();
+};
+
+}  // end of namespace mozilla::dom
+
+#endif